src/components/Modal/index.tsx

```diff
@@ -298,6 +305,13 @@ const Modal: React.FC<ModalProps> = (props) => {
         if (mounted.current) {
           setAnimatingIn(false);
         }
+        if (isLoading) {
+          // wait before showing modal
+          await waitMs(MS_TO_WAIT_BEFORE_SHOWING_LOADING_INDICATOR);
+          setShowModal(true);
```

Comment:
Can you wrap this to check if mounted is true first? If the modal disappears first, this `setShowModal` will cause a react warning.

```ts
if (mounted.current) {
  setShowModal(true);
}
```